spec/support/mailers.rb

```diff
@@ -0,0 +1,48 @@
+ActionMailer::Base.delivery_method = :test
+
+class GlobalIDArgument
```

Comment:
If you're moving these out of the spec can you nest them in a module to take them off the top level, e.g. `Support::Mailers`, alternatively use shared examples to define them in both specs

Reply:
I tried to use shared contexts, but Rubocop has complaints about that - https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Lint/ConstantDefinitionInBlock

I agree that it should not define classes within blocks. I also don't like defining them within `let` blocks as it complicates things.

I would go with the current solution without surrounding these classes with modules. And this is why:
1. The previous implementation would define these classes globally anyways. Yes, that would be conditionally, but **globally** - once runtime comes to `spec/rspec/rails/matchers/have_enqueued_mail_spec.rb` it defines these classes and they are available globally by all other specs. So moving this code into the "support" folder as is doesn't change anything from the runtime point of view, but actually fixes potential issues with the surprising dynamic appearance of these classes in the code base.
2. Namespacing would need to either rewrite many specs to use the full path to classes, e.g. instead of `TestMailer` it needs `Support::Mailers::TestMailer`. An `include Support::Mailers` on the top of the spec file would solve this point, but that would define class names globally dynamically again. An alternative solution could be redefining the class name (`def self.name`) for these classes, but that would look weird in general and like an over-engineering.
3. The codebase already has an example of similar classes defined globally - see `support/ar_classes.rb`

So, looks like defining these classes in namespaces would just complicate things and that's unnecessary. WDYT? Any specific reason for defining them within a namespace?

Reply:
The reason for my original comment is I don't want to accidentally pollute our root namespace making it seem like these are real classes, as you're moving them and creating churn, might as well fix that.

Reply:
That causes too many questions that seem not related to the PR. I found another solution: I will define the mailer inside the new test using `let` as it should have been done in the beginning. The issue of polluting root space remains though. It should be resolved separately.